src/types/bet_note.nr

```diff
@@ -30,26 +35,33 @@ use dep::aztec::{
 struct BetNote {
     owner: AztecAddress,
     bet_id: Field,
-    bet: bool,
+    bet: bool, // Heads is 0 (False). Tails is 1 (True).
+    escrow_randomness: Field,
     header: NoteHeader
 }
 
-global BET_NOTE_LEN: Field = 3;
+global BET_NOTE_LEN: Field = 4;
 
 impl BetNote {
     /////////////////////////////////////////////////////////////////////
     //                         Public methods                          //
     /////////////////////////////////////////////////////////////////////
 
     /// @notice Bet note constructor
+    /// @param _context The currenct context, used to push the bet_id nullifier in order to ensure uniqueness
     /// @param _owner The owner of the note
-    /// @param _bet_id The id of the bet (randomlty generated, by reusing the escrow randomness)
+    /// @param _bet_id The id of the bet (randomly generated, by reusing the escrow randomness)
     /// @param _bet The bet outcome given by the user (true or false, representing head or tail)
-    pub fn new(_owner: AztecAddress, _bet_id: Field, _bet: bool) -> Self {
+    pub fn new(_context: &mut PrivateContext, _owner: AztecAddress, _bet: bool, _bet_id: Field, _escrow_randomness: Field) -> Self {
+        // Nullify bet id
+        let nullifier =  dep::std::hash::pedersen_hash([_bet_id]);
```

Comment:
Shouldn't this be compute_nullifier (from this note impl)?

Reply:
Hmm, but doesn't it make the note as transient then (ie created and nullified within the same tx)? Not sure if there are additional considerations due to this

Reply:
Nvm, only the bet_id -> wdyt of moving this into another fn, "generate_bet_id" or smth? Shouldn't we check the bet_id isn't nullified before nullifying it?

Reply:
Didn't find a way to make the constrained nullification work. Added an unconstrained method to enable testing for whether the id was nullified or not, though. Also, moved the generation of the bet_id nullifier into a separate function.